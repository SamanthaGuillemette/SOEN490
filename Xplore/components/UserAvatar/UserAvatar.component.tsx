import { Avatar } from "react-native-paper";
import { View } from "../View";
import styles from "./UserAvatar.styles";
import { ShadowView } from "../ShadowView";
import { useThemeColor } from "../../hooks";
import { profileArray } from "../../assets/imageManager";

interface UserAvatarProps {
  index: number;
  size: 135 | 60 | 45 | 31;
}

export const UserAvatar = (props: UserAvatarProps) => {
  const { index, size } = props;
  const whiteBackground = useThemeColor("backgroundSecondary");
  let image = profileArray[index];

  return (
    <View style={styles.Avatar}>
      <ShadowView style={[styles.Image, { borderColor: whiteBackground }]}>
        <Avatar.Image size={size} source={image} />
      </ShadowView>
    </View>
  );
};

export default UserAvatar;
