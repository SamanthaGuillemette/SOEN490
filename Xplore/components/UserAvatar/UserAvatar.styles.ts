import { StyleSheet } from "react-native";
import { colors } from "../../constants";

export default StyleSheet.create({
  Avatar: {
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between",
  },
  Image: {
    borderRadius: 200,
    borderWidth: 2,
    borderColor: colors.light.backgroundSecondary,
  },
});
