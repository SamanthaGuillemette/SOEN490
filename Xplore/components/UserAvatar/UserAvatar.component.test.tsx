import React from "react";
import { cleanup } from "react-native-testing-library";
import renderer from "react-test-renderer";
import UserAvatar from "./UserAvatar.component";

afterAll(() => {
  cleanup();
});

describe("UserAvatar should render correctly", () => {
  it("should render user avatar correctly", () => {
    const avatar = renderer.create(<UserAvatar index={0} size={135} />).toJSON;
    expect(avatar).toMatchSnapshot();
  });
});
