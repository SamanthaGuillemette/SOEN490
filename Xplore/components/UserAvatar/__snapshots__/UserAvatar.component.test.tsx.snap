// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should render correctly should render user avatar correctly 1`] = `[Function]`;
